src/net_processing.cpp

```diff
@@ -2243,7 +2243,9 @@ void PeerManagerImpl::ProcessOrphanTx(std::set<uint256>& orphan_work_set)
             break;
         }
     }
-    m_mempool.check(&::ChainstateActive().CoinsTip());
+    CChainState& active_chainstate = m_chainman.ActiveChainstate();
+    CCoinsViewCache& active_coins_tip = active_chainstate.CoinsTip();
+    m_mempool.check(&active_coins_tip, active_chainstate.m_blockman.GetSpendHeight(active_coins_tip));
```

Comment:
Shouldn't an assert be added here and before the following invocations ? The `CCoinsViewCache` pointer is used throughout `CTxMemPool::check.

Reply:
Fixed in: 52c29d9aedcf8d141b75a6d923a5d5bea53f4079